src/test/java/net/oauth/jsontoken/JsonTokenParserTest.java

```diff
@@ -215,24 +403,68 @@ public void testPublicKey() throws Exception {
   }
 
   private boolean verifyTimeFrame(Instant issuedAt, Instant expiration) throws Exception {
+    JsonTokenParser parser = getJsonTokenParser();
+    JsonToken checkToken = getJsonTokenWithTimeRange(issuedAt, expiration);
+
+    try {
+      parser.verify(checkToken);
+      return true;
+    } catch (IllegalStateException e) {
+      return false;
+    }
+  }
+
+  private JsonToken getJsonTokenWithTimeRange(Instant issuedAt, Instant expiration) throws Exception {
     HmacSHA256Signer signer = new HmacSHA256Signer("google.com", "key2", SYMMETRIC_KEY);
     JsonToken token = new JsonToken(signer, clock);
     if (issuedAt != null) {
       token.setIssuedAt(issuedAt);
     }
+
     if (expiration != null) {
       token.setExpiration(expiration);
     }
-    token.setAudience("http://www.google.com");
 
-    JsonTokenParser parser = new JsonTokenParser(clock, locators, new IgnoreAudience());
-    JsonToken checkToken = parser.deserialize(token.serializeAndSign());
-    try {
-      parser.verify(checkToken);
-      return true;
-    } catch (IllegalStateException e) {
-      return false;
-    }
+    JsonToken checkToken = new JsonToken(
+        token.getHeader(),
+        token.getPayloadAsJsonObject(),
+        clock,
+        token.serializeAndSign()
+    );
+
+    return checkToken;
+  }
+
+  private JsonToken naiveDeserialize(String tokenString) {
+    assertTrue(
+        tokenString.equals(TOKEN_STRING_2PARTS)
+          || tokenString.equals(TOKEN_STRING_EMPTY_SIG)
+    );
```

Comment:
Is there any reason to restrict this method? Could we instead just generalize it and use it instead of parser.deserialize (to get the added benefit that the verify tests don't depend on the success of deserialize as you mentioned earlier).

Reply:
Since this method was only intended to be used with these two token strings, I added this additional check to make sure we don't use it for the wrong thing. However, it can be generalized. I have replaced instances of deserialize in the verify tests with this function